.github/workflows/build.yml

```
@@ -131,4 +137,58 @@ jobs:
+          key: ${{ runner.os }}-tests-unit-coverage-v3-${{ env.BRANCH_NAME }}-${{hashFiles('tests/unit/coverage/lcov.info')}}
+
+      - name: Save unit tests coverage report as artifact 💎
+        uses: actions/upload-artifact@v4
+        with:
+          name: unit-tests-coverage-report
+          path: ${{ env.UNIT_TESTS_COVERAGE_REPORTS_PATH }}
+          retention-days: 2
+
+      - name: Exit with error if unit tests failed 🚨
+        if: steps.unit-tests.outcome != 'success'
+        run: exit 1
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)